### PR TITLE
Prevent eslint from flipping out in npm test

### PR DESCRIPTION
**Pull request by irvinebroque**
This fixes the issue that @zoopoetics reported internally, where #21 caused `eslint-loader` to try to lint the compiled webpack output.

In addition, it fixes up eslint errors that were previously not being picked up.

The tradeoff here is that `npm run watch-tests` will no longer continuously watch for linter errors in the background. This will unfortunately be the case until we can get `eslint-loader` and `isparta-loader` to play nice together.

### CODE REVIEW

File: examples/area-and-line/index.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ require('../../src/index.scss');
 
 function getData() {
   var start = moment().startOf('week');
-  var end = moment(start).endOf('week');
```
Comment (irvinebroque) [inline examples/area-and-line/index.js]: eslint error, no-unused-var

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.git
+build
+lib
+node_modules
```
Comment (irvinebroque) [inline .eslintignore]: Don't lint these directories.

File: examples/area/index.js [hunk 3]
```diff
@@ -19,12 +18,11 @@ function getData() {
   return data;
 }
 
-var data = getData();
```
Comment (irvinebroque) [inline examples/area/index.js]: redeclare from `getData` scope above.

File: examples/area/index.js [hunk 4]
```diff
@@ -33,8 +31,8 @@ var groups = [{
 var width = 1024;
 React.render(
   <TimeseriesChart
-    height={Math.ceil(width * 0.75)}
```
Comment (irvinebroque) [inline examples/area/index.js]: Alphabetical prop ordering eslint error

File: webpack.config.js [hunk 5]
```diff
@@ -1,4 +1,3 @@
-var webpack = require('webpack');
```
Comment (irvinebroque) [inline webpack.config.js]: unused

File: package.json [hunk 6]
```diff
@@ -10,8 +10,8 @@
     "build": "webpack -p --progress --colors",
     "prepublish": "NODE_ENV=prepublish webpack --progress --colors",
     "examples": "webpack-dev-server --config examples/webpack.config.js --content-base examples",
-    "lint": "webpack -p --progress --colors",
-    "test": "karma start --single-run",
+    "lint": "eslint .",
+    "test": "karma start --single-run && eslint .",
```
Comment (irvinebroque) [inline package.json]: Runs them sequentially, rather than using eslint-loader in `karma.config.js`

**Comment by irvinebroque** [comment]
@zoopoetics - see above. I can revisit this tomorrow and see if I can get `eslint-loader` to play nice, but this fixes the immediate problem, even if it's more ideal to use the loader.

**Comment by zoopoetics** [comment]
@irvinebroque - I was doing some general maintenance stuff this afternoon and caught a lot of these. Looks like we were cosmically aligned.

**Pull request merged**